### PR TITLE
SOC-531 Update Weekly Digest to use new templates

### PR DESCRIPTION
**Pull request by jsutterfield**
Update the weekly digest to use the new template design.

Ticket: https://wikia-inc.atlassian.net/browse/SOC-531

### CODE REVIEW

File: extensions/wikia/GlobalWatchlist/GlobalWatchlistBot.class.php [hunk 1]
```diff
@@ -111,103 +107,21 @@ public function sendDigestToUser( $userID ) {
...
-				} else {
-					$aWikiDigest = [
-						'wikiName' => $oWikia->city_title,
-						'wikiLangCode' => $oWikia->city_lang,
-						'pages' => []
-					];
-				}
-			} // if
-
-			if ( in_array( $oResultRow->gwa_namespace, [ NS_BLOG_ARTICLE_TALK, NS_BLOG_ARTICLE ] ) ) {
```
Comment (jsutterfield) [inline extensions/wikia/GlobalWatchlist/GlobalWatchlistBot.class.php]: The old weekly digest email gave special treatment to blogs and article comments. It called them out in a separate section of the email. The new designs don't draw any distinction here, all changes (article, category, blogs. etc) are displayed in the same way.

I double checked with Cristina and she said that's the behavior we want, so this logic, and all the logic found in `makeBlogsList()` can be removed.

File: extensions/wikia/GlobalWatchlist/GlobalWatchlistBot.class.php [hunk 2]
```diff
@@ -295,221 +209,115 @@ private function checkIfSubscribedToWeeklyDigest( \User $user ) {
...
 
-			$sDigests .= "\n";
-		}
-		if ( $isDigestLimited ) {
-			$sDigests .= $this->getLocalizedMsg( 'globalwatchlist-see-more', $oUserLanguage ) . "\n";
-		}
-		$aEmailArgs = [
-			0 => ucfirst( $oUser->getName() ),
-			1 => ( $iPagesCount > 0 ) ? $sDigests : $this->getLocalizedMsg( 'globalwatchlist-no-page-found', $oUserLanguage ),
-			2 => ( $iBlogsCount > 0 ) ? $sDigestsBlogs : "",
```
Comment (garthwebb) [inline extensions/wikia/GlobalWatchlist/GlobalWatchlistBot.class.php]: Oh god this is terrible; why explicitly call out the keys only to make them numbered indexes?!  Either make a normal array or make meaningful names ...

File: extensions/wikia/Email/Controller/WeeklyDigestController.class.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	protected function getSubject() {
+		return $this->getMessage( 'emailext-weeklydigest-subject' )->text();
+	}
+
+	protected function getSummary() {
+		return $this->getMessage( 'emailext-weeklydigest-summary' )->text();
+	}
+
+	public function initEmail() {
+		$this->digestData = $this->request->getVal( 'digestData' );
```
Comment (garthwebb) [inline extensions/wikia/Email/Controller/WeeklyDigestController.class.php]: Two related notes; 1) I don't see a ```getEmailSpecificFormFields``` method here 2) The ```digestData``` parameter as is will be impossible to pass from the email tool :(

I don't have any super clever ideas, but maybe we could test whether ```digestData``` is a string or an array, and if its a string assume is JSON and decode it into an array. Then at least we could pass ```digestData``` via the tool, even if it is super awkward.

File: extensions/wikia/Email/EmailController.class.php [hunk 4]
```diff
@@ -648,7 +648,7 @@ private static function getLegendName() {
 	/**
 	 * A wrapper for wfMessage() which removes the possibility of messages being overridden in the MediaWiki namespace
 	 *
-	 * @return Message
+	 * @return \Message
```
Comment (garthwebb) [inline extensions/wikia/Email/EmailController.class.php]: Jinx!

File: extensions/wikia/Email/i18n/WeeklyDigest.i18n.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
+<?php
+$messages = array();
+
+$messages['en'] = array(
+	'emailext-weeklydigest-subject' => "See what's new on your favorite Wikia pages this week",
+	'emailext-weeklydigest-summary' => "Since you're last visit, Wikia Community members have made improvements to pages that you follow.",
```
Comment (mklucsarits) [inline extensions/wikia/Email/i18n/WeeklyDigest.i18n.php]: s/you're/your

File: extensions/wikia/Email/i18n/WeeklyDigest.i18n.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+);
+
+$messages['qqq'] = array(
+	'emailext-weeklydigest-subject' => 'Subject line Weekly Digest email. This email includes links to pages the user is following across Wikia which have been edited in the last week',
+	'emailext-weeklydigest-summary' => "Line telling the user that the links below are for pages which they follow and have been edited since they're last visit to the page",
```
Comment (mklucsarits) [inline extensions/wikia/Email/i18n/WeeklyDigest.i18n.php]: s/they're/their

File: extensions/wikia/Email/styles/weeklyDigest.css [hunk 7]
```diff
@@ -0,0 +1,24 @@
+h3 {
+	font-family: "Helvetica Neue", Arial, Sans-serif;
```
Comment (mklucsarits) [inline extensions/wikia/Email/styles/weeklyDigest.css]: Just FYI there's no need to put font names in quotes. Modern browsers don't care.

File: extensions/wikia/Email/EmailController.class.php (hunk 4 above)
Comment (mklucsarits) [inline extensions/wikia/Email/EmailController.class.php]: :D

File: extensions/wikia/Email/templates/greeting.mustache [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<div class="greeting">
+	<p>{{salutation}}</p>
+	<p>{{{summary}}}</p>
```
Comment (mklucsarits) [inline extensions/wikia/Email/templates/greeting.mustache]: Nothing wrong here, but just a heads up that there'll be a conflict with SOC-459 where Garth added ```strong``` tags around the summary.

File: extensions/wikia/Email/i18n/WeeklyDigest.i18n.php (hunk 6 above)
Comment (jsutterfield) [inline extensions/wikia/Email/i18n/WeeklyDigest.i18n.php]: Whaddaya speak English as ya first language or sumthin? :)

File: extensions/wikia/Email/styles/weeklyDigest.css (hunk 7 above)
Comment (jsutterfield) [inline extensions/wikia/Email/styles/weeklyDigest.css]: Huh, didn't know that. That's nifty.

File: extensions/wikia/Email/templates/greeting.mustache (hunk 8 above)
Comment (jsutterfield) [inline extensions/wikia/Email/templates/greeting.mustache]: Nice catch. Addressed here: https://github.com/Wikia/app/pull/7414/files#r31855194

File: extensions/wikia/Email/Controller/WeeklyDigestController.class.php (hunk 3 above)
Comment (jsutterfield) [inline extensions/wikia/Email/Controller/WeeklyDigestController.class.php]: Yea, I don't have any super clever ideas either. I think the JSON string approach is the best way for now. I'm also going to include a default value for that field so user's of Special:SendEmail can just send it off and see the digest without having to figure out how to insert json with the correct values into that field.

**Comment by garthwebb** [comment]
LGTM

**Pull request merged**